src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -151,102 +158,68 @@ class TezosNodeOperator(node: TezosRPCInterface)(implicit ec: ExecutionContext)
-    * decide when to stop the recursive descent on predecessors
-    *
-    * @param network    which Tezos network to go against
-    * @param current    The current metadata object to start descending
-    * @param minLevel   The lowest level for which to include results, unless a fork should be followed
-    * @param maxLevel   The highest level for which to include results
-    * @param followFork If the predecessor of the minLevel block appears to be on a fork, also capture the blocks on the fork.
-    * @param collected  The currently accumulated metadata so far
-    * @return           The metadata list, wrapped in a [[Future]]
+    * Get the blocks in a specified range.
+    * @param network Which tezos network to go against
+    * @param minLevel Minimum Block Level
+    * @param maxLevel MaximumBlockLevel
+    * @param startBlockHash If specified, start from the supplied block hash, otherwise, head of the chain.
+    * @param followFork If predecessor of the minLevel block appears to be on a fork, also capture the blocks on the fork.
+    * @return Blocks
     */
-  def fetchChainedMetadata(
-    network: String,
-    current: BlockMetadata,
-    minLevel: Int,
-    maxLevel: Int,
-    followFork: Boolean,
-    collected: List[BlockMetadata] = List.empty
-  ) : Future[List[BlockMetadata]] = {
-    val currentLevel = current.header.level
-    logger.info("Current metadata level is {}", currentLevel)
-    val reachedBottom = currentLevel == 0 && minLevel <= 0 //can we get further on? Why checking min?
-    val reachedMinNoForkin = currentLevel == minLevel && !followFork
-    val belowMinWithForking = currentLevel <= minLevel && followFork
-    val aboveMax = currentLevel > maxLevel
-    val aboveMin = currentLevel > minLevel
-
-    if(reachedBottom || reachedMinNoForkin)
-      Future.successful(current :: collected)
-    else if (belowMinWithForking)
-      TezosDatabaseOperations.blockExists(current.header.predecessor) flatMap { predecessorStored =>
-        if (predecessorStored)
-        Future.successful(current :: collected) //pred on db, we're done here
-      else
-        getMetadata(network, current.header.predecessor) flatMap { pred =>
-          fetchChainedMetadata(network, pred, minLevel, maxLevel, followFork, current :: collected)
-        }
-      }
-    else if (aboveMax)
-      //skip this
-      getMetadata(network, current.header.predecessor) flatMap { pred =>
-        fetchChainedMetadata(network, pred, minLevel, maxLevel, followFork, collected)
-      }
-    else if (aboveMin)
-      getMetadata(network, current.header.predecessor) flatMap { pred =>
-        fetchChainedMetadata(network, pred, minLevel, maxLevel, followFork, current :: collected)
-      }
-    else
-      Future.successful(List.empty)
-
+  def getBlocks(network: String, minLevel: Int, maxLevel: Int,
+                startBlockHash: Option[String], followFork: Boolean): Future[List[Block]] = {
+    val hash = startBlockHash.getOrElse("head")
```

Comment:
Okay, refactored so instead of startBlockHash being an Option[String], it will be a string that's an optional parameter with "head" as the default value.